src/App.tsx

```diff
@@ -16,36 +18,55 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [filter, setFilter] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const filterHandler = (action: string) => {
+    switch (action) {
+      case 'alphabet':
+        setFilter('alphabet');
+        break;
+      case 'length':
+        setFilter('length');
+        break;
+      case 'reverse':
+        setIsReversed(prev => !prev);
+        break;
+      case 'reset':
+        setFilter('');
+        setGoods(goodsFromServer);
+        setIsReversed(false);
+        break;
+      default:
+        break;
+    }
+  };
```

Comment:
[CODE STYLE] - The function's name `filterHandler` should start with a verb to clearly indicate what it does. Consider renaming it to something like `handleFilter` to follow this best practice.